### PR TITLE
improved screen reader support for both dice roller examples

### PR DESCRIPTION
**Pull request by micahgodbolt**
Fixes #https://github.com/microsoft/FluidFramework/issues/3620

Screen reader now skips reading the dice character, and reads the dice value in the string "Left dice value is five" (and also "Right dice value is 5")

### CODE REVIEW

**Comment by micahgodbolt** [comment]
@SamBroner hey sam, do you a moment to look at this?

File: src/fluidObjects/diceRoller/DiceRollerView.tsx [hunk 1]
```diff
@@ -29,20 +30,16 @@ export const DiceRollerView = (props: IAppProps) => {
             // When the view dismounts remove the listener to avoid memory leaks
             props.model.off("diceRolled", onDiceRolled);
         };
-    // eslint-disable-next-line react-hooks/exhaustive-deps
-    }, [diceValue]);
+        // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, [props]);
```
Comment (SamBroner) [inline src/fluidObjects/diceRoller/DiceRollerView.tsx:34]: We were just doing this wrong?

**Review by SamBroner — APPROVED** [review]
Looks good!

File: src/fluidObjects/diceRoller/DiceRollerView.tsx (hunk 1 above)
Comment (micahgodbolt) [reply src/fluidObjects/diceRoller/DiceRollerView.tsx:34]: how often do we want to re-evaluate this useEffect? Every time the diceValue changes? Or when the component props change?

**Pull request merged**